ethers-providers/src/transports/common.rs

```diff
@@ -94,6 +94,15 @@ impl ResponseData<serde_json::Value> {
     }
 }
 
+/// Basic or bearer authentication in http or websocket transport
+///
+/// Use to inject username and password or an auth token into requests
+#[derive(Clone, Debug)]
+pub enum Authorization {
+    Basic(String, String),
```

Comment:
can we add constructors for both variants so that a user knows that `Basic` is name, password?